### PR TITLE
feat(nrql_conditions): add new fields 'expiration' and 'signal'

### PR DESCRIPTION
**Pull request by sanderblue**
Facilitates: [newrelic/terraform-provider-newrelic #810](https://github.com/newrelic/terraform-provider-newrelic/issues/810)

### CODE REVIEW

File: pkg/alerts/nrql_conditions.go [hunk 1]
```diff
@@ -6,6 +6,40 @@ import (
...
+	CloseViolationsOnExpiration bool `json:"closeViolationsOnExpiration"`
+	ExpirationDuration          int  `json:"expirationDuration"`
+	OpenViolationOnExpiration   bool `json:"openViolationOnExpiration"`
+}
+
+// AlertsNrqlConditionSignal - Configuration that defines the signal that the NRQL condition will use to evaluate.
+type AlertsNrqlConditionSignal struct {
+	EvaluationOffset int              `json:"evaluationOffset"`
+	FillOption       AlertsFillOption `json:"fillOption"`
+	FillValue        float64          `json:"fillValue"`
```
Comment (ahornerr) [inline pkg/alerts/nrql_conditions.go:21]: Fill value is nullable in the API, suggest making it a pointer

File: pkg/alerts/nrql_conditions.go [hunk 2]
```diff
@@ -6,6 +6,40 @@ import (
 	"github.com/newrelic/newrelic-client-go/pkg/errors"
 )
 
+// AlertsNrqlConditionExpiration - **Preview access:** These fields may be viewed and set, but will not be active until the release date.
+// Settings for how violations are opened or closed when a signal expires.
+type AlertsNrqlConditionExpiration struct {
+	CloseViolationsOnExpiration bool `json:"closeViolationsOnExpiration"`
+	ExpirationDuration          int  `json:"expirationDuration"`
```
Comment (ahornerr) [inline pkg/alerts/nrql_conditions.go:13]: Expiration duration is nullable in the API, suggest making it a pointer

File: pkg/alerts/nrql_conditions.go (hunk 1 above)
Comment (sanderblue) [reply pkg/alerts/nrql_conditions.go:21]: roger that

File: pkg/alerts/nrql_conditions.go (hunk 2 above)
Comment (sanderblue) [reply pkg/alerts/nrql_conditions.go:13]: 10-4

**Review by ahornerr — DISMISSED** [review]

**Review by ahornerr — DISMISSED** [review]

File: pkg/alerts/nrql_conditions.go [hunk 3]
```diff
@@ -6,6 +6,42 @@ import (
 	"github.com/newrelic/newrelic-client-go/pkg/errors"
 )
 
+// AlertsNrqlConditionExpiration - **Preview access:** These fields may be viewed and set, but will not be active until the release date.
+// Settings for how violations are opened or closed when a signal expires.
+// nolint:golint
+type AlertsNrqlConditionExpiration struct {
+	ExpirationDuration          *int `json:"expirationDuration"`
```
Comment (zlesnr) [inline pkg/alerts/nrql_conditions.go:13]: `omitempty`?

Comment (zlesnr) [reply pkg/alerts/nrql_conditions.go:13]: Same question for the other pointers in `AlertsNrqlConditionSignal`.

**Review by zlesnr — COMMENTED** [review]
This looks good to me.  Just a question about when to include the `omitempty` on the pointer fields.

File: pkg/alerts/nrql_conditions.go [hunk 4]
```diff
@@ -565,6 +603,16 @@ const (
 		}
 		type
 		violationTimeLimit
+		expiration {
+      closeViolationsOnExpiration
```
Comment (ctrombley) [inline pkg/alerts/nrql_conditions.go:607]: nit: fix indentation?

**Review by ctrombley — APPROVED** [review]
🌮

File: pkg/alerts/nrql_conditions.go (hunk 3 above)
Comment (ahornerr) [reply pkg/alerts/nrql_conditions.go:13]: The nuance of the GraphQL service, merging of previous/new fields, optional values, default values in the API, and backwards compatible fields requires a somewhat specific combination of pointers and inclusion/exclusion of `omitempty`.

For the most part, these have to be pointers in order to allow the Terraform to be backwards compatible (not requiring users to begin setting these options until they're ready). GraphQL differentiates between nil with `omitempty` (Javascript `undefined`) versus nil without `omitempty` (Javascript `null`). In order to support options like `fillValue`, it is required some fields must have the ability to be null.

File: pkg/alerts/nrql_conditions.go (hunk 4 above)
Comment (sanderblue) [reply pkg/alerts/nrql_conditions.go:607]: Not sure I see what you're seeing 🤔

Comment (sanderblue) [reply pkg/alerts/nrql_conditions.go:607]: Ah, it's my chrome extension thats hiding it. Must be a copy and paste thing.

**Review by zlesnr — DISMISSED** [review]

**Comment by zlesnr** [comment]
🐐 ☁️

**Review by zlesnr — APPROVED** [review]
More 🐐

**Pull request merged**